telethon/telegram_client.py

```diff
@@ -476,6 +476,55 @@ def get_message_history(self,
 
         return total_messages, result.messages, entities
 
+    def get_all_participants(self,entity):
+        """
+        Returns all the partcipants for the specified entity.
+
+        :param entity:      The entity from whom to retrieve the participants
+
+        :return: A list of participants for the given entity
+        """
+
+        def update_participant_set(participants, participant):
```

Comment:
There's absolutely no need for this function, sets don't allow duplicated. However, you may still get duplicates (even with this function) because the library returns *new* user objects (which are stored somewhere else in memory so they're not the same). You should maybe keep a dictionary `{id: user}` and add by `d[user.id] = user`, then return `list(d.values())`.

Reply:
sweet i'll use a dictionary instead